pkg/appender/ingest.go

```diff
@@ -250,8 +258,14 @@ func (mc *MetricsCache) postMetricUpdates(metric *MetricState) {
 		} else if sent {
 			metric.setState(storeStateUpdate)
 		}
-		if !sent && metric.store.samplesQueueLength() == 0 {
-			metric.setState(storeStateReady)
+		if !sent {
```

Comment:
we need to also do some handling for the other `writeChunks` call, no?
in `ingest.go:353`